src/app/workflow/dag/wdl-viewer/wdl-viewer.component.ts

```diff
@@ -0,0 +1,110 @@
+import { GA4GHFilesService } from '../../../shared/ga4gh-files/ga4gh-files.service';
+import { ExtendedWorkflow } from '../../../shared/models/ExtendedWorkflow';
+import { WorkflowQuery } from '../../../shared/state/workflow.query';
+import { ToolDescriptor, WorkflowsService, WorkflowVersion } from '../../../shared/swagger';
+import { WdlViewerService, WdlViewerPipeline } from './wdl-viewer.service';
+
+@Component({
+  selector: 'app-wdl-viewer',
+  templateUrl: './wdl-viewer.html',
+  styleUrls: ['./wdl-viewer.component.scss'],
+  providers: [WdlViewerService],
+  encapsulation: ViewEncapsulation.None,
+})
+export class WdlViewerComponent implements AfterViewInit, OnInit, OnDestroy {
+
+  @Input() workflow: ExtendedWorkflow;
+  @Input() expanded: boolean;
+  @Input() descriptorType: ToolDescriptor.TypeEnum;
+  @Input() set selectedVersion(value: WorkflowVersion) {
+    this.loading = true;
+    this.onVersionChange(value);
+  }
+  @ViewChild('diagram') diagram: ElementRef;
+  @ViewChild('exportSVG') exportSVG: ElementRef;
+
+  public pipelineBuilderResult$: Observable<any>;
+  public errorMessage;
+  public loading = false;
+  public wdlViewerError = false;
+  protected entryType: ('tool' | 'workflow') = 'workflow';
+  protected ngUnsubscribe: Subject<{}> = new Subject();
+  private version: WorkflowVersion;
+  private visualizer: any;
+
+  constructor(private wdlViewerService: WdlViewerService, public fileService: FileService, protected gA4GHFilesService: GA4GHFilesService,
+              protected workflowsService: WorkflowsService, private renderer: Renderer2, private workflowQuery: WorkflowQuery) {
+  }
+
+  ngOnInit() {
+    this.loading = true;
+  }
+
+  ngAfterViewInit() {
+    this.visualizer = new pipeline.Visualizer(this.diagram.nativeElement, false);
+
+    // Retrieve all files for this workflow from Ga4ghFiles entity Store
+    this.wdlViewerService.getFiles(ToolDescriptor.TypeEnum.WDL).pipe(filter(file => file != null), takeUntil(this.ngUnsubscribe))
+      .subscribe(files => {
+        if (files && files.length > 0) {
+          this.pipelineBuilderResult$ = this.wdlViewerService.create(files, this.workflow, this.version);
+
+          // Create the Epam pipeline builder visualization and attach the result to the DOM element
+          this.pipelineBuilderResult$.pipe(takeUntil(this.ngUnsubscribe), finalize(() => this.loading = false))
+            .subscribe((res: WdlViewerPipeline) => {
+                this.visualizer.attachTo(res.model[0]);
+                this.wdlViewerError = false;
+              },
+              (error) => {
+                this.errorMessage = error || 'Unknown Error';
+                this.wdlViewerError = true;
+                this.diagram.nativeElement.remove();
+              });
+        }
+      });
+  }
+
+  ngOnDestroy() {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  onVersionChange(value: WorkflowVersion) {
+    if (this.visualizer) {
```

Comment:
Do you need a `this.loading = true` here? It doesn't seem like you'll get the progress bar when you change the version.

Reply:
Actually, I added that when the change in the `selectedVersion` variable is detected

https://github.com/dockstore/dockstore-ui2/blob/2b6135cd3a1ea2a09801e39290b34b1e94c57898/src/app/workflow/dag/wdl-viewer/wdl-viewer.component.ts#L41-L44

Reply:
👍